pkg/networkmanager/firewall-client.js

```diff
@@ -167,21 +178,39 @@ function fetchServiceInfos(services) {
         if (firewall.services[service])
             return firewall.services[service];
 
+        let info;
         return firewalld_dbus.call('/org/fedoraproject/FirewallD1',
                                    'org.fedoraproject.FirewallD1',
                                    'getServiceSettings', [service])
                 .then(reply => {
-                    const [ , name, description, ports ] = reply[0];
-
-                    let info = {
+                    const [ , name, description, ports, , , , ] = reply[0];
+                    info = {
                         id: service,
                         name: name,
                         description: description,
-                        ports: ports.map(p => ({ port: p[0], protocol: p[1] }))
+                        ports: ports.map(p => ({ port: p[0], protocol: p[1] })),
+                        includes: [],
                     };
 
                     firewall.services[service] = info;
+                    return firewalld_dbus.call('/org/fedoraproject/FirewallD1/config',
+                                               'org.fedoraproject.FirewallD1.config',
+                                               'getServiceByName', [service]);
+                })
+                .then(path => firewalld_dbus.call(path[0],
+                                                  'org.fedoraproject.FirewallD1.config.service',
+                                                  'getSettings2', []))
```

Comment:
Follow-up: Start with this call, and if it fails, fall back to the old one and set includes to empty. Saves one call on new firewalld, and on older versions there's no change (one succeeding and one failing call).

Reply:
Right, that would make a lot more sense. I'll do it as a followup, this one has been in the pipeline too long :p